### PR TITLE
Misc fixes 2

### PR DESCRIPTION
**Pull request by murrant**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Device/Sensor.php [hunk 1]
```diff
@@ -424,7 +429,7 @@ protected static function processSensorData($sensors, $prefetch)
                 $sensor_value = (cast_number($sensor_value) * $sensor['sensor_multiplier']);
             }
 
-            $sensor_data[$sensor['sensor_id']] = $sensor_value;
+            $sensor_data[$sensor['sensor_id'] ?? 0] = $sensor_value; // if no sensor_id, this is a new sensor, place at the first index
```
Comment (Jellyfrog) [inline LibreNMS/Device/Sensor.php:432]: Since it's in a loop, it could overwrite index 0 multiple times?

Comment (murrant) [reply LibreNMS/Device/Sensor.php:432]: Yeah, I had a hard time understanding this code.   The only time sensor_id is null is during discovery of a new (unsaved) sensor.
After fetching the data the discovery runs current($data) (Arr::first essentially) so it only takes one result out of the array.

Comment (murrant) [reply LibreNMS/Device/Sensor.php:432]: The code needs to be tossed, but it is currently working :/

File: LibreNMS/Device/Sensor.php [hunk 2]
```diff
@@ -400,7 +405,7 @@ protected static function processSensorData($sensors, $prefetch)
 
             // if no data set null and continue to the next sensor
             if (empty($data)) {
-                $data[$sensor['sensor_id']] = null;
+                $data[$sensor['sensor_id'] ?? 0] = null; // if no sensor_id, this is a new sensor, place at the first index
                 continue;
             }
```
Comment (Jellyfrog) [inline LibreNMS/Device/Sensor.php:410]: I don't get what this code even does, since `$data` is overwritten on each iteration.

Comment (murrant) [reply LibreNMS/Device/Sensor.php:410]: Pulled this into a separate PR.  I'll remove it from here.

**Review by Jellyfrog — APPROVED** [review]

**Comment by bbnoc** [comment]
Could it be this update broke this: https://github.com/librenms/librenms/pull/13248 ?

Got issues with non-unicode characters 14. august.

includes/polling/ports.inc.php

image

**Comment by kimhaak** [comment]
> Could it be this update broke this: #13248 ?
> 
> Got issues with non-unicode characters 14. august.
> 
> includes/polling/ports.inc.php
> 
> image

This was me...was logged in with different account...

**Comment by murrant** [comment]
@bbnoc @kimhaak yes, already fixed in daily

**Comment by kimhaak** [comment]
Thanks - Still seeing it on "type" 

image

**Comment by murrant** [comment]
Type never inferred the encoding.   I've never seen an issue reported for it.  Odd stuff that some device is using non-unicode characters for type.

**Comment by kimhaak** [comment]
This has to be some old "stuck" stuff - it does not appear in the discovery or poller capture.

In there, the vlan is type l2vlan. 

Where can i find the type of the vlan to remove it and see what is discovered after that?

**Comment by murrant** [comment]
@kimhaak database

**Comment by kimhaak** [comment]
@murrant - Thanks, but everything was ok there...is there any other places "Type" is used?

**Pull request merged**